launchable/test_runners/pytest.py

```diff
@@ -92,4 +104,74 @@ def _pytest_formatter(test_path):
 split_subset = launchable.CommonSplitSubsetImpls(
     __name__, formatter=_pytest_formatter).split_subset()
 
-record_tests = launchable.CommonRecordTestImpls(__name__).report_files()
+
+@click.option('--json', 'json_report', help="use JSON report files produced by pytest-dev/pytest-reportlog", is_flag=True)
+@click.argument('source_roots', required=True, nargs=-1)
+@launchable.record.tests
+def record_tests(client, json_report, source_roots):
+
+    ext = "json" if json_report else "xml"
+    for root in source_roots:
+        match = False
+        for t in glob.iglob(root, recursive=True):
+            match = True
+            if os.path.isdir(t):
+                client.scan(t, "*.{}".format(ext))
+            else:
+                client.report(t)
+
+        if not match:
+            click.echo("No matches found: {}".format(root), err=True)
+            return
+
+    if json_report:
+        client.parse_func = PytestJSONReportParser(client).parse_func
+
+    client.run()
+
+
+"""
+If you want to use --json option, please install pytest-dev/pytest-reportlog. (https://github.com/pytest-dev/pytest-reportlog)
+
+Usage
+
+```
+$ pip install -U pytest-reportlog
+$ pytest --report-log report.json
+$ launchable record tests --json report.json
+```
+"""
+
+
+class PytestJSONReportParser:
+    def __init__(self, client):
+        self.client = client
+
+    def parse_func(self, report_file: str) -> Generator[CaseEventType, None, None]:
+        with open(report_file, 'r') as json_file:
```

Comment:
[nits]
you can load json file 

```python
with open(report_file, 'r') as json_file:
    d = json.load(json_file)
```

`d` will be a dict from json file

Reply:
wait, do you like to validate each line of json file?
https://github.com/launchableinc/cli/pull/376/files#diff-f30bf261afdbc96299251de2735d7eea4b7456235c83a70cd3352140dd66bc5bR154

Reply:
Yes, right. pytest-dev/pytest-reportlog produces this format https://en.wikipedia.org/wiki/JSON_streaming